browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -197,12 +197,20 @@ EngineFlags ShouldBlockRequestOnTaskRunner(
       url::Origin::CreateFromNormalizedTuple("https", "youtube.com", 80),
       net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
 
+  std::string rewritten_url;
+
   SCOPED_UMA_HISTOGRAM_TIMER("Brave.Adblock.ShouldBlockRequest");
   g_brave_browser_process->ad_block_service()->ShouldStartRequest(
       url_to_check, ctx->resource_type, source_host,
       ctx->aggressive_blocking || force_aggressive,
       &previous_result.did_match_rule, &previous_result.did_match_exception,
-      &previous_result.did_match_important, &ctx->mock_data_url);
+      &previous_result.did_match_important, &ctx->mock_data_url,
+      &rewritten_url);
+
+  if (ctx->method == "GET" || ctx->method == "HEAD" ||
+      ctx->method == "OPTIONS") {
+    ctx->new_url_spec = rewritten_url;
```

Comment:
check validity of the url?

Reply:
Would it also be worth checking that the scheme is HTTP/HTTPS?

Reply:
updated